src/components/input-time/input-time.tsx

```diff
@@ -0,0 +1,872 @@
+
+  /** When true, the icon is flipped in RTL. */
+  @Prop({ reflect: true }) iconFlipRtl = false;
+
+  /**
+   * aria-label for the hour input
+   *
+   * @default "Hour"
+   */
+  @Prop() intlHour = TEXT.hour;
+
+  /**
+   * aria-label for the meridiem (am/pm) input
+   *
+   * @default "AM/PM"
+   */
+  @Prop() intlMeridiem = TEXT.meridiem;
+
+  /**
+   * aria-label for the minute input
+   *
+   * @default "Minute"
+   */
+  @Prop() intlMinute = TEXT.minute;
+
+  /**
+   * aria-label for the second input
+   *
+   * @default "Second"
+   */
+  @Prop() intlSecond = TEXT.second;
+
+  /**
+   * BCP 47 language tag for desired language and country format
+   *
+   * @internal
+   */
+  @Prop({ attribute: "lang", mutable: true }) locale: string =
+    document.documentElement.lang || navigator.language || "en";
+
+  @Watch("locale")
+  localeWatcher(newLocale: string): void {
+    this.hourCycle = getLocaleHourCycle(newLocale);
+    this.setValue({ newValue: this.value, context: "lang" });
+  }
+
+  /** The name of the time input */
+  @Prop() name: string;
+
+  /**
+   * Specifies the Unicode numeral system used by the component for localization.
+   */
+  @Prop({ reflect: true }) numberingSystem: NumberingSystem;
+
+  /**
+   * When true, still focusable but controls are gone and the value cannot be modified.
+   *
+   * @mdn [readOnly](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly)
+   */
+  @Prop() readonly = false;
+
+  /**
+   * When true, makes the component required for form-submission.
+   *
+   * @internal
+   */
+  @Prop({ reflect: true }) required = false;
+
+  /** The scale (size) of the time input */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /** number (seconds) that specifies the granularity that the value must adhere to */
+  @Prop() step = 60;
+
+  /** The selected time in UTC (always 24-hour format) */
+  @Prop({ mutable: true }) value: string = null;
+
+  @Watch("value")
+  valueWatcher(newValue: string): void {
+    if (!this.userChangedValue) {
+      this.setValue({ newValue, context: "direct" });
+    }
+    this.userChangedValue = false;
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  private activeEl: HTMLSpanElement;
+
+  defaultValue: InputTime["value"];
+
+  formEl: HTMLFormElement;
+
+  private hourEl: HTMLSpanElement;
+
+  labelEl: HTMLCalciteLabelElement;
+
+  private meridiemEl: HTMLSpanElement;
+
+  private meridiemOrder: number;
+
+  private minuteEl: HTMLSpanElement;
+
+  private previousEmittedValue: string;
+
+  previousValue: string;
+
+  private secondEl: HTMLSpanElement;
+
+  private userChangedValue = false;
+
+  // --------------------------------------------------------------------------
+  //
+  //  State
+  //
+  // --------------------------------------------------------------------------
+
```

Comment:
Is there a way to simplify these State props? It seems like we only need the localized state or regular state, not both as updating one could drive the other.